source/Classes/1.DscResources/08.nxScript.ps1

```diff
@@ -0,0 +1,120 @@
+[DscResource()]
+class nxScript
+{
+    [DscProperty(Key)]
+    [System.String] $GetScript
+
+    [DscProperty(Key)]
+    [System.String] $TestScript
+
+    [DscProperty(Key)]
+    [System.String] $SetScript
+
+    [DscProperty()]
+    [Reason[]] $Reasons
+
+    [nxScript] Get()
+    {
+        $scriptBlock = [System.Management.Automation.ScriptBlock]::Create($this.GetScript)
+        $invokeScriptResult = $this.InvokeScript($scriptBlock)
+        $isValid = $this.TestGetScriptOutput($invokeScriptResult)
+
+        $currentState = [nxScript]::new()
+        $currentState.GetScript = $this.GetScript
+        $currentState.TestScript = $this.TestScript
+        $currentState.SetScript = $this.SetScript
+        if ($isValid)
+        {
+            $currentState.Reasons = $invokeScriptResult.Reasons
+        }
+
+        return $currentState
+    }
+
+    [bool] Test()
+    {
+        $scriptBlock = [System.Management.Automation.ScriptBlock]::Create($this.TestScript)
+        $invokeScriptResult = $this.InvokeScript($scriptBlock)
+
+        if ($invokeScriptResult -is [System.Management.Automation.ErrorRecord])
+        {
+            Write-Verbose -Message "TestScript returned an error."
```

Comment:
Can you print the error as part of this message? Just so that the customer is never sitting there wondering what the error is?